### ISSUE TITLE
Format selection doesn't fix indentation

### ISSUE THREAD

**Issue by jrieken** [microsoft/vscode#105998]
So it looks like that may have been an issue with my formatter (Prettier). Disabled that and used the default one and got it working for JS files. 

However, I'm not seeing this work reliably with JSON files. Some format commands don't work (`Format Selection` and `Format Modified Lines`) but `Format Document` does work 🤷‍♂️:

![Kapture 2020-09-02 at 08 13 16](https://user-images.githubusercontent.com/35271042/92001999-398def80-ecf4-11ea-8b11-3d5d6d603292.gif)

_Originally posted by @misolori in https://github.com/microsoft/vscode/issues/105800#issuecomment-685804555_

**Comment by aeschli** [comment]
Very unclear to me what the expect behavior of range format regarding the initial indentation. 
Can we assume the content above is correctly formatted and indented?
Right now JSON builds on the original indent of the selected range.

TypeScript seems to look at the full file
``
{
{
var x; // selected line
}
}
```
=>
``
{
{
        var x; // selected line
}
}
```

**Comment by jrieken** [comment]
Yeah, I think the general approach is to format the whole document but to only generate text edits for nodes inside the selected range.
